pkg/operator/watchdog/proc.go

```diff
@@ -0,0 +1,52 @@
+package watchdog
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// GetPidByName find the process name specified by name and return the PID of that process.
+// If the process is not found, the bool is false.
+// NOTE: This require container with shared process namespace (if run as side-car).
+func GetPidByName(name string) (int, bool, error) {
```

Comment:
FindPid?

Reply:
It is by using name... golang already have funcs to get pid, but not by process name... `FindPidByName` ?

Reply:
FindPidByName is good